localstack/services/sqs/provider.py

```diff
@@ -0,0 +1,1243 @@
+        actions: ActionNameList,
+    ) -> None:
+        queue = self._require_queue_by_url(queue_url)
+        self._assert_permission(context, queue)
+
+        self._validate_actions(actions)
+
+        for account_id in aws_account_ids:
+            for action in actions:
+                queue.permissions.add(Permission(label, account_id, action))
+
+    def remove_permission(self, context: RequestContext, queue_url: String, label: String) -> None:
+        queue = self._require_queue_by_url(queue_url)
+        self._assert_permission(context, queue)
+
+        candidate = None
+        for perm in queue.permissions:
+            if perm.label == label:
+                candidate = perm
+                break
+        if candidate:
+            queue.permissions.remove(candidate)
+
+    def _create_message_attributes(
+        self,
+        context: RequestContext,
+        message_system_attributes: MessageBodySystemAttributeMap = None,
+    ) -> Dict[MessageSystemAttributeName, str]:
+        result: Dict[MessageSystemAttributeName, str] = {
+            MessageSystemAttributeName.SenderId: context.account_id,
+            MessageSystemAttributeName.SentTimestamp: str(now()),
+        }
+
+        if message_system_attributes is not None:
+            for attr in message_system_attributes:
+                result[attr] = message_system_attributes[attr]["StringValue"]
+
+        return result
+
+    def _validate_queue_attributes(self, attributes: QueueAttributeMap):
+        valid = [k[1] for k in inspect.getmembers(QueueAttributeName)]
+
+        for k in attributes.keys():
+            if k not in valid:
+                raise InvalidAttributeName("Unknown Attribute %s" % k)
+
+    def _validate_actions(self, actions: ActionNameList):
+        service = load_service(service=self.service, version=self.version)
+        # FIXME: this is a bit of a heuristic as it will also include actions like "ListQueues" which is not
+        #  associated with an action on a queue
+        valid = list(service.operation_names)
+        valid.append("*")
+
+        for action in actions:
+            if action not in valid:
+                raise InvalidParameterValue(
+                    f"Value SQS:{action} for parameter ActionName is invalid. Reason: Please refer to the appropriate "
+                    "WSDL for a list of valid actions. "
+                )
+
+    def _assert_permission(self, context: RequestContext, queue: SqsQueue):
+        action = context.operation.name
+        account_id = context.account_id
+
+        if account_id == queue.owner:
+            return
+
+        for permission in queue.permissions:
+            if permission.account_id != account_id:
+                continue
+            if permission.action == "*":
+                return
+            if permission.action == action:
+                return
+
+        raise CommonServiceException("AccessDeniedException", "Not allowed (TODO: correct message)")
```

Comment:
i think this code is never actually reached right now anyway. so maybe you can just move the TODO to a comment for now. seems unnecessary to go through the exercise of testing it :-) especially since the message payload of errors are not really super critical for clients.